dockstore-webservice/src/main/java/io/dockstore/webservice/helpers/statelisteners/ElasticListener.java

```diff
@@ -464,24 +412,64 @@ private static Stream<Version> versionStream(Set<Version> versions) {
         }
     }
 
-    private static Set<Version> cloneWorkflowVersion(final Set<Version> originalWorkflowVersions) {
+    private static Set<Version> detachVersions(final Set<Version> originalWorkflowVersions, final Version defaultVersion) {
         Set<Version> detachedVersions = new HashSet<>();
         originalWorkflowVersions.forEach(workflowVersion -> {
             Version detachedVersion = workflowVersion.createEmptyVersion();
-            detachedVersion.setDescriptionAndDescriptionSource(workflowVersion.getDescription(), workflowVersion.getDescriptionSource());
             detachedVersion.setInputFileFormats(new TreeSet<>(workflowVersion.getInputFileFormats()));
             detachedVersion.setOutputFileFormats(new TreeSet<>(workflowVersion.getOutputFileFormats()));
             detachedVersion.setName(workflowVersion.getName());
             detachedVersion.setReference(workflowVersion.getReference());
-            SortedSet<SourceFile> sourceFiles = workflowVersion.getSourceFiles();
-            sourceFiles.forEach(sourceFile -> detachedVersion.addSourceFile(SourceFile.copy(sourceFile)));
-            detachedVersion.updateVerified();
+            // Only include the description and sourcefiles in the default version
+            if (workflowVersion == defaultVersion) {
+                detachedVersion.setDescriptionAndDescriptionSource(workflowVersion.getDescription(), workflowVersion.getDescriptionSource());
+                SortedSet<SourceFile> sourceFiles = workflowVersion.getSourceFiles();
+                sourceFiles.forEach(sourceFile -> detachedVersion.addSourceFile(SourceFile.copy(sourceFile)));
+            }
             detachedVersions.add(detachedVersion);
         });
         return detachedVersions;
     }
 
-    private static Workflow detachWorkflow(Workflow detachedWorkflow, Workflow workflow) {
+    private static void copyEntryProperties(Entry detachedEntry, Entry entry) {
```

Comment:
Was like this already, but this could easily be missed when adding new properties since this lives in this listener.
Perhaps a copy constructor-like method living in Entry would be less missable 
https://www.geeksforgeeks.org/copy-constructor-in-cpp/

Reply:
> Was like this already, but this could easily be missed when adding new properties since this lives in this listener. Perhaps a copy constructor-like method living in Entry would be less missable https://www.geeksforgeeks.org/copy-constructor-in-cpp/

I pondered something like this whilst writing the PR, and I'm on the fence...

If we put the logic in the entries, you have a clean `createDetachedEntry` method that's more likely to stay in sync (like you mention).

However, how to copy the properties is really `ElasticListener`'s decision, so we're putting its logic into each entry, and there's a higher possibility that future devs get confused and use these methods for purposes they're not supposed to.

Curious what others think.

Reply:
:thought_balloon:  
What distinguishes the elastic search copy from other potential copies, I would guess that it would be the absence of confidential data or data that is too large? (maybe we skip things like ownership)

Reply:
I lean towards keeping it in `ElasticListener` because it seems specific to ES and it's nice to look at it all in one place, but I can see the advantages of having it in the Entry (ex: if we have a new entry type, it would be obvious that you need to override `Entry.createDetachedEntry`). I don't have super strong feelings about it, so I would be okay with either approach

Reply:
Ok, the reviewers are 3-1 in favor this code remaining in ElasticListener, so we'll keep it the way it is.

Reply:
Think I'm ok with that along with a follow-up ticket to investigate a way of having both, as in visibility that something is concurring in both ElasticListener and in Entry when developers add new entries. 
Possibly something along the lines of having `@JsonView` annotations in Entry but defining the view here https://www.baeldung.com/jackson-json-view-annotation